compendium/ConfigurationAdmin/src/ConfigurationAdminImpl.cpp

```diff
@@ -704,39 +700,46 @@ namespace cppmicroservices
                         });
 
                     auto const factoryPid = getFactoryPid(pid);
-                    if (factoryPid.empty())
+                    if (!factoryPid.empty())
```

Comment:
you can further constrain the visibility of `factoryPid` by turning this if statement into one with an initializer, e.g.:

```
if(auto const factoryPid = getFactoryPid(pid); !factoryPid.empty())
....
```